File: 728.自除数.py
#
# @lc app=leetcode.cn id=728 lang=python3
#
# [728] 自除数
#

# @lc code=start
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        def valid(number):
            for num in str(number):
                if num == "0" or number % int(num) != 0:
                    return False
            return True
        res = []
        for num in range(left, right + 1):
            if valid(num):
                res.append(num)
        return res
# @lc code=end

File: 728.自除数.go
/*
 * @lc app=leetcode.cn id=728 lang=golang
 *
 * [728] 自除数
 */

// @lc code=start
func selfDividingNumbers(left int, right int) (res []int) {
	for num := left; num < right + 1; num++ {
		if valid(num) {
			res = append(res, num)
		}
	}
	return
}


func valid(number int) bool {
	for _, curr := range strconv.Itoa(number) {
		currInt := int(curr) - int('0')
		if currInt == 0 || number % currInt != 0 {
			return false
		}
	}
	return true
}
// @lc code=end

